components/tools/OmeroJava/test/integration/chown/PermissionsTest.java

```diff
@@ -366,22 +561,101 @@ public void testChownAnnotationReadAnnotate(boolean isDataOwner, boolean isAdmin
         assertOwnedBy(image, recipient);
         assertOwnedBy(ownerAnnotations, importer);
         assertOwnedBy(otherAnnotations, annotator);
+
+        /* chown the first tag set */
+        chown = Requests.chown().target(tagsets.get(0)).toUser(recipient.userId).build();
+
+        switch (option) {
+        case NONE:
+            break;
+        case INCLUDE:
+            chown.childOptions = Collections.singletonList(Requests.option().includeType("Annotation").build());
+            break;
+        case EXCLUDE:
+            chown.childOptions = Collections.singletonList(Requests.option().excludeType("Annotation").build());
+            break;
+        case BOTH:
+            chown.childOptions = Collections.singletonList(Requests.option().includeType("Annotation")
+                                                                            .excludeType("Annotation").build());
+            break;
+        default:
+            Assert.fail("unexpected option for chown");
+        }
+        doChange(client, factory, chown, isExpectSuccess);
+
+        if (!isExpectSuccess) {
+            return;
+        }
+
+        /* check that the tag set is transferred and the other remains owned by original owner */
+        assertOwnedBy(tagsets.get(0), recipient);
+        assertOwnedBy(tagsets.get(1), annotator);
+
+        /* check that only the expected tags are transferred */
+        switch (option) {
+        case NONE:
+            assertOwnedBy(tags.get(0), annotator);
+            assertOwnedBy(tags.get(1), annotator);
+            assertOwnedBy(tags.get(2), annotator);
+            break;
+        case BOTH:
+            /* include overrides exclude */
+        case INCLUDE:
+            assertOwnedBy(tags.get(0), recipient);
+            assertOwnedBy(tags.get(1), recipient);
+            assertOwnedBy(tags.get(2), annotator);
+            break;
+        case EXCLUDE:
+            assertOwnedBy(tags.get(0), annotator);
+            assertOwnedBy(tags.get(1), annotator);
+            assertOwnedBy(tags.get(2), annotator);
+            /* transfer the tag that is not in the second tag set */
+            init(chowner);
+            chown = Requests.chown().target(tags.get(0)).toUser(recipient.userId).build();
+            doChange(client, factory, chown, isExpectSuccess);
+            if (!isExpectSuccess) {
+                return;
```

Comment:
can omit this expect success stuff, including from above `doChange`, as mentioned above